### ISSUE TITLE
Invalid floating type error when compiling shader

### ISSUE THREAD

**Issue by mbouchard2** [microsoft/DirectXShaderCompiler#7915]
**Description**
Currently getting following error when compiling shader:
error: Invalid floating type
UNREACHABLE executed at C:\__w\1\s\DXC\include\llvm/IR/Type.h:174!

**Steps to Reproduce**
https://godbolt.org/z/GMafYrfj9

If we just compile with ps_6_8 instead:
https://godbolt.org/z/eK91GGPe8

**Actual Behavior**
Since this actual shader do compile with shader model 6.8, I would expect to compile on 6.9 as well otherwise, the error should point to proper line number.

**Environment**
- DXC version 1.8.2505.32
- Host Operating System Windows 11 Version 23H2 (OS Build 22631.6060)

**Comment by V-FEXrt** [comment]
The assert is being raised here:
https://github.com/microsoft/DirectXShaderCompiler/blob/66793582ebbc3ef9c3bec245d30895f017b2a5cd/include/llvm/IR/Type.h#L174

`getTypeID` is expectedly a `vector` in the SM6.9 case which isn't hitting any of the branches. We could potentially update that function directly however the call site may also be the right place to update.

https://github.com/microsoft/DirectXShaderCompiler/blob/66793582ebbc3ef9c3bec245d30895f017b2a5cd/lib/Transforms/Scalar/Float2Int.cpp#L416

Even in the SM6.8 case, this transform isn't applied because a `float` can't losslessly fit into the `uint` and it just bottoms out in doing nothing. The SM6.9 case should also just bottom out in doing nothing but exactly _how_ to do nothing is unclear to me.

One option is to subtly rewrite the code to

```C++
    if (MinBW > 64) {
      DEBUG(dbgs() << "F2I: Value requires more than 64 bits to represent!\n");
      continue;
    }
    unsigned MaxRepresentableBits
      = APFloat::semanticsPrecision(ConvertedToTy->getFltSemantics()) - 1;
    if (MinBW > MaxRepresentableBits) {
      DEBUG(dbgs() << "F2I: Value not guaranteed to be representable!\n");
      continue;
    }
```

which checks `MinBW` before calling `getFltSemantics`. MinBW for a `float2` is `65` so it fails the check and early exits as expected. This seems like a _complete_ solution since all bigger vectors would have larger `MinBW` and all smaller "vectors" are just the scalar case which already works correctly. I'm having a hard time convincing myself this is a "correct" solution vs a solution that just happens to work though.

Willing to hear anyone's opinions

**Comment by tex3d** [comment]
I don't understand why vectors should come into play at all, unless there's another bug. Why would you expect to add up the mantissa bits from multiple scalars in a vector to see if it's representable by a scalar int type? Am I missing something?
